### PR TITLE
fix(batch): computeEnvironments is now required for JobQueue

### PR DESCRIPTION
**Pull request by andrestone**
## Commit Message
fix(batch): `computeEnvironments` is now required for JobQueue (#6616)

Fixes: #6615

BREAKING CHANGE: `computeEnvironments` is now required

## End Commit Message

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-batch/lib/job-queue.ts [hunk 1]
```diff
@@ -106,24 +105,16 @@ export class JobQueue extends Resource implements IJobQueue {
...
           order: cp.order,
-        } as CfnJobQueue.ComputeEnvironmentOrderProperty))
-        : [
-            {
-              // Get an AWS Managed Compute Environment
-              computeEnvironment: new ComputeEnvironment(this, 'Resource-Batch-Compute-Environment').computeEnvironmentArn,
-              order: 1,
-            },
-          ],
+        } as CfnJobQueue.ComputeEnvironmentOrderProperty)),
```
Comment (ayush987goyal) [inline packages/@aws-cdk/aws-batch/lib/job-queue.ts]: Can we have a validation on the length (not-zero) of the `computeEnvironment` prop?

Comment (andrestone) [reply packages/@aws-cdk/aws-batch/lib/job-queue.ts]: Yep!

File: packages/@aws-cdk/aws-batch/lib/job-queue.ts [hunk 2]
```diff
@@ -106,24 +105,20 @@ export class JobQueue extends Resource implements IJobQueue {
...
   public readonly jobQueueName: string;
 
-  constructor(scope: Construct, id: string, props: JobQueueProps = {}) {
+  constructor(scope: Construct, id: string, props: JobQueueProps) {
     super(scope, id, {
       physicalName: props.jobQueueName,
     });
 
+    if (props.computeEnvironments.length === 0) {
+      throw new Error('It\'s not allowed to pass an empty array of JobQueueComputeEnvironment elements.');
```
Comment (iliapolo) [inline packages/@aws-cdk/aws-batch/lib/job-queue.ts]: `array` and `JobQueueComputeEnvironment` is redundant here

```suggestion
`computeEnvironments` must be non-empty
```

Comment (andrestone) [reply packages/@aws-cdk/aws-batch/lib/job-queue.ts]: Done!

**Comment by andrestone** [comment]
Done, I guess! :D

**Comment by iliapolo** [comment]
> Done, I guess! :D

Almost :) did the final thouchups

Thanks!

**Pull request merged**